Sources/Sentry/SentryTransaction.m

```diff
@@ -112,11 +117,43 @@ - (void)setOperation:(NSString *)operation
     [self.spanContext setOperation:operation];
 }
 
+- (SentrySpan *)startChildWithOperation:(NSString *)operation
+{
+    return [self startChildWithOperation:operation description:nil];
+}
+
+- (SentrySpan *)startChildWithOperation:(NSString *)operation
+                            description:(nullable NSString *)description
+{
+    return [self startChildWithParentId:self.spanId operation:operation description:description];
+}
+
+- (SentrySpan *)startChildWithParentId:(SentrySpanId *)parentId
+                             operation:(NSString *)operation
+                           description:(nullable NSString *)description
+{
+    SentrySpan *span = [[SentrySpan alloc] initWithTransaction:self
+                                                       traceId:self.traceId
+                                                      parentId:parentId];
+    span.operation = operation;
+    span.spanDescription = description;
+    span.sampled = self.isSampled;
+    @synchronized(self.spans) {
+        [self.spans addObject:span];
+    }
+    return span;
+}
+
 - (NSDictionary<NSString *, id> *)serialize
 {
     NSMutableDictionary<NSString *, id> *serializedData =
         [[NSMutableDictionary alloc] initWithDictionary:[super serialize]];
-    serializedData[@"spans"] = @[];
+
+    NSMutableArray *spans = [[NSMutableArray alloc] init];
+    for (SentrySpan *span in self.spans) {
+        [spans addObject:[span serialize]];
+    }
+    serializedData[@"spans"] = spans;
```

Comment:
`h`: We need to make that thread-safe as well.

Reply:
It's true that this needs to be thread safe but it comes back to the fact that Transaction is an event, which isn't (and will never be) thread-safe. So we need to discuss what we do with this.

Lets discuss this Friday please, @mitsuhiko will be there

Reply:
Thursday though